src/lib/PageTracker.js

```diff
@@ -0,0 +1,410 @@
+            const { id: fileId, file_version: { id: fileVersionId } = null } = this.file;
+            // Add the current event to the events Array.
+            this.events.push({
+                userId: Number(this.userId),
+                timestamp: pageViewUTCTimestamp,
+                payload: {
+                    page: {
+                        number: this.currentPage,
+                        viewMs: timeToBeReported,
+                    },
+                    sessionId: this.sessionId,
+                },
+                target: {
+                    id: Number(fileId),
+                    length: this.fileLength,
+                    ownerEId: this.fileOwnerEId,
+                    type: 'file',
+                    versionId: Number(fileVersionId),
+                },
+            });
+            if (this.previewEventReported) {
+                this.emit(REPORT_ACI, { events: [...this.events], type: 'pageView' });
+                // Clear the events array
+                this.events.length = 0;
+            }
+        }
+        this.startTime = Date.now();
+    }
+
+    /**
+     * Updates file preview container reference
+     *
+     * @private
+     * @param {HTMLElement} container - File Preview container
+     * @return {void}
+     */
+    setContainer(container) {
+        this.container = container;
+    }
+
+    /**
+     *  Set current page data and length from the event payload
+     *
+     * @private
+     * @param {Object} eventData - Event payload
+     * @return {void}
+     */
+    setCurrentPage({ currentPage = 0 }) {
+        this.currentPage = currentPage;
+    }
+
+    /**
+     *  Set file length data
+     *
+     * @private
+     * @param {Object} eventData - Event payload
+     * @return {void}
+     */
+    setFileLength({ numPages = 0 }) {
+        this.fileLength = numPages;
+    }
+
+    /**
+     * Updates the flag that indicates if the Access Stats PREVIEW event was successfully reported.
+     * If the request fails, we won't send additional insights because we don't want to have
+     * data inconsistency
+     *
+     * @private
+     * @param {boolean} reported
+     * @return {void}
+     */
+    setPreviewEventReported(reported) {
+        this.previewEventReported = reported;
+    }
+
+    /**
+     *  Set content insights options
+     *
+     * @private
+     * @param {Object} options - Content Insights options
+     * @return {void}
+     */
+    setOptions(options) {
+        // Set User and file information
+        ({
+            ownerEId: this.fileOwnerEId = '',
+            userEId: this.userEId = '',
+            userId: this.userId,
+            config: { isActive: this.active = false } = {},
+        } = options);
+    }
+
+    /**
+     *  Update content insights options
+     *
+     * @private
+     * @param {Object} options - Content Insights options
+     * @return {void}
+     */
+    updateOptions(options = {}) {
```

Comment:
This seems similar to `init`. When would we need to update the options without reinitializing?

Reply:
I'm thinking of a case where the content-insights options request was completed after the initialization of the tracker.